### PR TITLE
DOC: Styler errors PR08 and PR09

### LINKED ISSUE #28656: DOC: Styler errors PR08 and PR09
```python
python scripts/validate_docstrings.py pandas.io.formats.style.Styler --errors=PR08,PR09

```
#### Problem description
returns:
```
11 Errors found:
	Parameters {table_attributes} not documented
	Parameter "data" has no description
	Parameter "precision" description should start with a capital letter
	Parameter "precision" description should finish with "."
	Parameter "table_styles" description should start with a capital letter
	Parameter "table_styles" description should finish with "."
	Parameter "uuid" description should start with a capital letter
	Parameter "uuid" description should finish with "."
	Parameter "caption" description should start with a capital letter
	Parameter "caption" description should finish with "."
	Missing description for See Also "DataFrame.style" reference
2 Warnings found:
	No extended summary found
	No examples section found
```

This is a subset of error messages from #28602 and #25232. Part of the work for #27977

### PR DESCRIPTION
**Pull request by bhuvanakundumani**
-  closes #28656
-  tests added/ passed - Checked with 
```
./scripts/validate_docstrings.py pandas.io.formats.style.Styler --errors=PR08,PR09
```
- passes `black pandas`
- passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- whatsnew entry - not needed

### CODE REVIEW

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -77,6 +81,8 @@ class Styler:
     See Also
     --------
     DataFrame.style
+        Returns a Styler object containing methods for building
```
Comment (WillAyd) [inline pandas/io/formats/style.py]: I don't think this belongs in the `See Also` section - is this addressing some issue?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style.py (hunk 1 above)
Comment (bhuvanakundumani) [reply pandas/io/formats/style.py]: Had this error - Missing description for See Also "DataFrame.style" reference. Hence added this  line of description

Comment (WillAyd) [reply pandas/io/formats/style.py]: Oh OK thanks for clarifying. I think this is an issue with the validator as the format *should* match what you see in the docsring guide (specifically small description separated by a colon):

https://python-sprints.github.io/pandas/guide/pandas_docstring.html#section-5-see-also

Comment (bhuvanakundumani) [reply pandas/io/formats/style.py]: Made the changes as requested.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@bhuvanakundumani Thanks a lot !

**Comment by bhuvanakundumani** [comment]
> @bhuvanakundumani Thanks a lot !

Thanks @jorisvandenbossche .

**Comment by xaviripper0345** [comment]
Thanks

**Pull request merged**